src/t8_forest_catalyst.cxx

```diff
@@ -0,0 +1,455 @@
+  vtkNew < vtkWedge > prism;
+  vtkNew < vtkTetra > tet;
+
+  vtkNew < vtkQuadraticEdge > quadraticedge;
+  vtkNew < vtkQuadraticTriangle > quadratictri;
+  vtkNew < vtkQuadraticQuad > quadraticquad;
+  vtkNew < vtkQuadraticTetra > quadratictet;
+  vtkNew < vtkQuadraticHexahedron > quadratichexa;
+  vtkNew < vtkQuadraticWedge > quadraticprism;
+
+  /* 
+   * The cellTypes Array stores the element types as integers(see vtk doc).
+   */
+  num_elements = t8_forest_get_local_num_elements (forest);
+  int                *cellTypes = T8_ALLOC (int, num_elements);
+
+  /*
+   * We need the vertex coords array to be of the 
+   * correct dim. Since it is always the same
+   * in one mesh, we take the dim of one element.
+   * We add 1 if we look at a vertext(dim=0) because 
+   * an array of size 0 is not allowed. 
+   * Then we allocate memory, because we do not know
+   * beforehand how many entries the array needs.
+   */
+
+  /*
+   * We have to define the vtkTypeInt64Array that hold 
+   * metadata if wanted. 
+   */
+
+  t8_vtk_gloidx_array_type_t *vtk_treeid = t8_vtk_gloidx_array_type_t::New ();
+  t8_vtk_gloidx_array_type_t *vtk_mpirank =
+    t8_vtk_gloidx_array_type_t::New ();
+  t8_vtk_gloidx_array_type_t *vtk_level = t8_vtk_gloidx_array_type_t::New ();
+  t8_vtk_gloidx_array_type_t *vtk_element_id =
+    t8_vtk_gloidx_array_type_t::New ();
+
+/*
+ * We need the dataArray for writing double valued user defined data in the vtu files.
+ * We want to write num_data many timesteps/arrays.
+ * We need num_data many vtkDoubleArrays, so we need to allocate storage.
+ * Later we call the constructor with: dataArrays[idata]=vtkDoubleArray::New()
+ */
+  vtkDoubleArray    **dataArrays;
+  dataArrays = T8_ALLOC (vtkDoubleArray *, num_data);
+
+  cmesh = t8_forest_get_cmesh (forest);
+  /* We iterate over all local trees*/
+  for (itree = 0; itree < t8_forest_get_num_local_trees (forest); itree++) {
+/* 
+ * We get the current tree, the scheme for this tree
+ * and the number of elements in this tree. We need the vertices of
+ * the tree to get the coordinates of the elements later. We need
+ * the number of elements in this tree to iterate over all of them.
+ */
+    t8_eclass_scheme_c *scheme =
+      t8_forest_get_eclass_scheme (forest, t8_forest_get_tree_class (forest,
+                                                                  itree));
+    t8_locidx_t         elems_in_tree =
+      t8_forest_get_tree_num_elements (forest, itree);
+    t8_locidx_t         offset =
+      t8_forest_get_tree_element_offset (forest, itree);
+    /* We iterate over all elements in the tree */
+    /* Compute the global tree id */
+    gtreeid = t8_forest_global_tree_id (forest, itree);
+    for (ielement = 0; ielement < elems_in_tree; ielement++) {
+      t8_element_t       *element =
+        t8_forest_get_element_in_tree (forest, itree, ielement);
+      T8_ASSERT (element != NULL);
+      vtkSmartPointer < vtkCell > pvtkCell = NULL;
```

Comment:
You construct a smart pointer for every element in this loop. Couldn't you move this line outside the loop and reuse this pointer for every element?